cpp/src/gandiva/projector.cc

```diff
@@ -83,6 +100,21 @@ Status Projector::Make(SchemaPtr schema, const ExpressionVector& exprs,
   std::unique_ptr<LLVMGenerator> llvm_gen;
   ARROW_RETURN_NOT_OK(LLVMGenerator::Make(configuration, is_cached, &llvm_gen));
 
+  if (!is_cached && sec_cache != nullptr) {
+    std::string key = std::string(Engine::ToString()) + cache_key.ToString();
```

Comment:
Have a separate method to get the sec cache key..which both projector and filter calls..also maybe add a delimiter char between the engine & cache_key.

Reply:
also add a comment there why cpu details are required